java/src/jmri/jmrit/display/layoutEditor/TransitCreationTool.java

```diff
@@ -76,9 +78,31 @@ public Transit createTransit() throws JmriException {
             log.error("Unable to create transit {}", transitName);
             throw new JmriException(Bundle.getMessage("TCTErrorUnableToCreate", transitName));
         }
-
+        jmri.SignalMastLogicManager smlm = InstanceManager.getDefault(jmri.SignalMastLogicManager.class);
+        int seqNo = 1;
+        // Add stub block section if applicable
+        SignalMastLogic smlForFirstMast = smlm.getSignalMastLogic((SignalMast) list.get(0));
+        if (smlForFirstMast != null) {
+            LayoutBlock layoutBlock = smlForFirstMast.getFacingBlock();
+            if (layoutBlock.getNumberOfNeighbours() == 1) {
+                // A stub track block has one neighbor
+                SectionManager sectionManager = InstanceManager.getDefault(SectionManager.class);
+                for (Section section : sectionManager.getNamedBeanSet()) {
+                    // Look for a user defined section that has one block that matches the layout block
+                    if (section.getSectionType() == Section.USERDEFINED) {
+                        if (section.getNumBlocks() == 1 && section.getEntryBlock().equals(layoutBlock.getBlock())) {
+                            t.addTransitSection(new jmri.TransitSection(section, seqNo, Section.FORWARD));
+                            seqNo++;
+                            break;
+                        }
+                    }
+                }
+                if (seqNo == 1) {
+                    log.warn("Unable to find a stub block section for {}", layoutBlock.getDisplayName());
+                }
+            }
+        }
         if (list.get(0) instanceof SignalMast) {
```

Comment:
The new block should be after the instanceof statement.  This insures that the list has at least one entry and that entry is a signal mast.